src/test/java/seedu/address/logic/commands/flashcardcommands/AddFlashcardSetCommandTest.java

```diff
@@ -0,0 +1,189 @@
+package seedu.address.logic.commands.flashcardcommands;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.FlashcardModel;
+import seedu.address.model.flashcard.Flashcard;
+import seedu.address.model.flashcard.FlashcardSet;
+import seedu.address.model.systemlevelmodel.ReadOnlyFlashcardBank;
+import seedu.address.testutil.FlashcardSetBuilder;
+
+public class AddFlashcardSetCommandTest {
+
+    @Test
+    public void constructor_nullFlashcardSet_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddFlashcardSetCommand(null));
+    }
+
+    @Test
+    public void execute_flashcardSetAcceptedByModel_addSuccess() throws Exception {
+        AddFlashcardSetCommandTest.ModelStubAcceptingFlsetAdded modelStub =
+                new AddFlashcardSetCommandTest.ModelStubAcceptingFlsetAdded();
+        FlashcardSet validFlset = new FlashcardSetBuilder().build();
+
+        CommandResult result = new AddFlashcardSetCommand(validFlset).execute(modelStub);
+
+        assertEquals(String.format(AddFlashcardSetCommand.MESSAGE_SUCCESS, validFlset), result.getFeedbackToUser());
+        assertEquals(Arrays.asList(validFlset), modelStub.flsetsAdded);
+    }
+
+    @Test
+    public void execute_duplicateFlashcardSet_throwsCommandException() {
+        FlashcardSet validFlset = new FlashcardSetBuilder().build();
+        AddFlashcardSetCommand addCommand = new AddFlashcardSetCommand(validFlset);
+        AddFlashcardSetCommandTest.ModelStubWithFlashcardSet modelStub =
+                new AddFlashcardSetCommandTest.ModelStubWithFlashcardSet(validFlset);
+
+        assertThrows(CommandException.class, AddFlashcardSetCommand.MESSAGE_DUPLICATE_FLASHCARD_SET, (
+        ) -> addCommand.execute(modelStub));
+    }
+
+    @Test
+    public void equals() {
+        FlashcardSet physics = new FlashcardSetBuilder().build();
+        FlashcardSet economics = new FlashcardSetBuilder().withFlashcardSetName("Economics").build();
```

Comment:
Minor stuff, we should avoid Magic String as much as possible. There is `VALID_FLSET_NAME_ECONOMICS` in the `FlashcardBankCommandTestUtil`

Reply:
Noted!